src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/GenerateMethodIndentService.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Razor;
+internal class GenerateMethodIndentService
+{
+    /// <summary>
+    ///  Adds indenting to the method.
+    /// </summary>
+    /// <param name="method">
+    ///  The method to add indenting to. The method should be marked with '0' where an indent is wanted and a '1' where an additional level of indent is wanted.
+    /// </param>
+    /// <param name="indentSize">The size of the indent.</param>
+    /// <param name="startingIndent">The size of the any existing indent.</param>
+    /// <returns>The indented method.</returns>
+    public static string AddIndentation(string method, int indentSize, int startingIndent)
+    {
+        var indent = string.Empty;
+        for (var i = 0; i < startingIndent + indentSize; i++)
+        {
+            indent += " ";
```

Comment:
Once you're using `RazorLSPOptions`  it will make more sense, but indentation could be in tabs, not spaces, so this is not quite complex enough. There is other code in `FormattingContext` (https://github.com/dotnet/razor/blob/main/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingContext.cs#L223 and the next couple of methods) that deal with this. Might be good to extract those to common helper methods?